src/main/java/org/jabref/gui/preferences/NameFormatterTabView.java

```diff
@@ -0,0 +1,104 @@
+        this.viewModel = nameFormatterTabViewModel;
+
+        formatterList.setEditable(true);
+
+        formatterNameColumn.setSortable(true);
+        formatterNameColumn.setReorderable(false);
+        formatterNameColumn.setCellValueFactory(new PropertyValueFactory<>("name"));
+        formatterNameColumn.setCellFactory(TextFieldTableCell.forTableColumn());
+        formatterNameColumn.setOnEditCommit(
+                (TableColumn.CellEditEvent<NameFormatterItemModel, String> event) -> {
+                    event.getTableView().getItems().get(
+                            event.getTablePosition().getRow())
+                            .setName(event.getNewValue());
+                });
+
+        formatterStringColumn.setSortable(true);
+        formatterStringColumn.setReorderable(false);
+        formatterStringColumn.setCellValueFactory(new PropertyValueFactory<>("format"));
+        formatterStringColumn.setCellFactory(TextFieldTableCell.forTableColumn());
+        formatterStringColumn.setOnEditCommit(
+                (TableColumn.CellEditEvent<NameFormatterItemModel, String> event) -> {
+                    event.getTableView().getItems().get(
+                            event.getTablePosition().getRow())
+                            .setFormat(event.getNewValue());
+                });
+
+        actionsColumn.setSortable(false);
+        actionsColumn.setReorderable(false);
+        actionsColumn.setCellValueFactory(new PropertyValueFactory<>("name"));
+        new ValueTableCellFactory<NameFormatterItemModel, String>()
+                .withGraphic(name -> IconTheme.JabRefIcons.DELETE_ENTRY.getGraphicNode())
+                .withTooltip(name -> Localization.lang("Remove") + " " + name)
+                .withOnMouseClickedEvent(item -> evt -> {
+                    nameFormatterTabViewModel.removeFormatter(formatterList.getFocusModel().getFocusedItem());
+                })
+                .install(actionsColumn);
+
+        formatterList.setOnKeyPressed(event -> {
+            if (event.getCode() == KeyCode.DELETE) {
+                nameFormatterTabViewModel.removeFormatter(formatterList.getSelectionModel().getSelectedItem());
+            }
+        });
+
+        formatterList.itemsProperty().bindBidirectional(nameFormatterTabViewModel.formatterListProperty());
+
+        addFormatterName.textProperty().bindBidirectional(nameFormatterTabViewModel.addFormatterNameProperty());
+        addFormatterString.textProperty().bindBidirectional(nameFormatterTabViewModel.addFormatterStringProperty());
+
+        ActionFactory actionFactory = new ActionFactory(Globals.getKeyPrefs());
+        actionFactory.configureIconButton(StandardActions.HELP_NAME_FORMATTER, new HelpAction(HelpFile.CUSTOM_EXPORTS_NAME_FORMATTER), formatterHelp);
+        actionFactory.configureIconButton(StandardActions.NAME_FORMATTER_ADD, new SimpleCommand() {
```

Comment:
You can add all the button information directly in the fxml file similar to the following:
https://github.com/JabRef/jabref/blob/3ebc59a764687cb68a6d550c99dce1c686bce174/src/main/java/org/jabref/gui/entryeditor/EntryEditor.fxml#L18-L25